docs/examples/schema_custom.py

```diff
@@ -1,7 +1,7 @@
 # output-json
 import json
 from pydantic import BaseModel
-from pydantic.schema import schema
+from pydantic.json_schema import schema
```

Comment:
Will openapi discriminator be supported?
If so then should this be openapi_schema?

Reply:
Please see #4666 - JSON Schema 2020-12 is equivalent to OpenAPI 3.1, so that's what we're supporting.

Reply:
So I read that and it did not say if discriminator is supported
per https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#properties
Openapi schema is a superset of json schema because it includes additional keywords like discriminator.
Will openapi discriminator be supported?

Reply:
Maybe @Relequestual can give some guidance on this, ideally we would support discriminator or some equivalent but I'd also like to stick to vanilla JSON Schema 2020-12, any ideas?

Reply:
2020-12 JSON Schema supports `dialects` and `vocabularies` and OpenAPI have created their own dialect on top of the standard JSON Schema dialect to extend the usage of four additional keywords. `xml`, `discriminator`, `externalDocs` and `example`.  As long as you support dialects and vocabularies, any extension should be loaded correctly and validate. 

https://github.com/OAI/OpenAPI-Specification/blob/main/schemas/v3.1/dialect/base.schema.json

https://json-schema.org/understanding-json-schema/reference/schema.html#:~:text=A%20version%20of%20JSON%20Schema,new%20dialect%20of%20JSON%20Schema.

cc @handrews

Reply:
well, in that case, if you want to support a user wanting to use `discriminator`, a dialect which supports that keyword should be used. In the case of OpenAPI, if they are using OAS3.1.x, the use of the OAS dialect should be preferred. I'm not familiar with another dialect using this keyword. Otherwise, creating a new dialect on the fly with the custom keywords could be another option.

Reply:
Yes, you could generate schemas with:

```JSON
{
    "$schema": "https://spec.openapis.org/oas/3.1/dialect/base",
    ...
}
```
instead of the standard 2020-12 dialect meta-schema, which includes all of standard 2020-12 plus the four [OpenAPI dialect extension keywords](https://spec.openapis.org/oas/latest.html#dialectSchemaId): `discriminator`, `xml`, `externalDocs`, and `example` (singular, deprecated in favor of the standard `examples`, plural).

Of those four, `discriminator` might be the only one that's relevant to you, but it is used fairly heavily in OpenAPI.  It will likely be removed in OpenAPI 4 in favor of a standard JSON Schema keyword or keywords that will work a bit differently.  `discriminator` doesn't quite fit the standard JSON Schema processing model.